python/dlisio/plumbing/errorhandler.py

```diff
@@ -4,10 +4,124 @@
 from .. import core
 
 class ErrorHandler(core.error_handler):
-    """
-    TODO ErrorHandler
-    """
+    """ Defines rules about error handling
+
+    Many .dlis files happen to be not compliant with specification or simply
+    broken. This class gives user some control over handling of such files.
+
+    For example:
+
+    * Users can throw on specification violations which dlisio doesn't consider
+      critical.
+    * Users can escape errors if they want to read as much data as possible.
+
+    Error handler does not cover all the warnings and exceptions.
+    It is mostly intended to be used for data processing, not for data
+    presentation. Thus it would be applicable, for example
+
+    * on data load and parsing in :py:func:`dlisio.load()`,
+      :py:func:`dlisio.logicalfile.load()` and
+      :py:func:`dlisio.plumbing.Frame.curves()`
+    * on object access :py:func:`dlisio.logicalfile.find()`,
+      :py:func:`dlisio.logicalfile.object()`
+    * on attribute access :py:attr:`dlisio.plumbing.Frame.description`
+
+    User still gets default behavior on other occasions, like warnings on
+    missing SUL or exceptions in :py:func:`dlisio.plumbing.Frame.curves()` in
+    case of invalid channel/frame relationships. Some of these might be get
+    implemented in the future.
+
+    Please also note that error handle doesn't redefine issues categories, it
+    only changes default behavior.
+
+    By supplying :py:attr:`Actions.RAISE` for lesser violations, user can
+    strengthen the behavior and throw even on minor issues.
+
+    By supplying :py:attr:`Actions.LOG_ERROR` for critical issues, user relaxes
+    the behavior, thus allowing errors escape.
+
+    By default only CRITICAL violations would be thrown as errors, everything
+    else will be logged.
```

Comment:
If anything, these are examples. However if you do something akin the critical example above, then this is sorta backed into that, so not need to spell it out explicitly